source/gui/nvdaControls.py

```diff
@@ -355,3 +354,63 @@ def onSliderChar(self, evt):
 			evt.Skip()
 			return
 		self.SetValue(newValue)
+
+
+class TabbableScrolledPanel(scrolledpanel.ScrolledPanel):
+	"""
+	This class was created to ensure a ScrolledPanel scrolls to nested children of the panel when navigating
+	with tabs (#12224). A PR to wxPython implementing this fix can be tracked on
+	https://github.com/wxWidgets/Phoenix/pull/1950
+	"""
+	def GetChildRectRelativeToSelf(self, child: wx.Window) -> wx.Rect:
+		"""
+		window.GetRect returns the size of a window, and its position relative to its parent.
+		When calculating ScrollChildIntoView, the position relative to its parent is not relevant unless the
+		parent is the ScrolledPanel itself. Instead, calculate the position relative to scrolledPanel
+		"""
+		cr = child.GetScreenRect()
+		spr = self.GetScreenPosition()
+		return wx.Rect(cr.x - spr.x, cr.y - spr.y, cr.width, cr.height)
+
+	def ScrollChildIntoView(self, child: wx.Window) -> None:
+		"""
+		Uses the same logic as super().ScrollChildIntoView(self, child)
+		except cr = self.GetChildRectRelativeToSelf(child) instead of cr = GetRect()
+		"""
+		sppu_x, sppu_y = self.GetScrollPixelsPerUnit()
+		vs_x, vs_y = self.GetViewStart()
+		cr = self.GetChildRectRelativeToSelf(child)
+		clntsz = self.GetClientSize()
+		new_vs_x, new_vs_y = -1, -1

```

Comment:
I would love it if there was a way to avoid having to duplicate the majority of this function.

Reply:
how does this new logic look? I am not a fan of overriding this function but it makes the actual change that we are doing clearer.